packages/react-hooks/src/hooks/debounced.ts

```diff
@@ -0,0 +1,29 @@
+import {useEffect, useRef, useState} from 'react';
+
+export const DELAY_COMMIT_SETTING_CHANGE = 500;
+
+export function useDebouncedValue<T>(
+  value: T,
+  delay = DELAY_COMMIT_SETTING_CHANGE,
+) {
+  const [state, setState] = useState<T>(value);
+  const updateRef = useRef<number>();
```

Comment:
Do we really need to keep the timeout in the ref?
If the `value` changes, the effect returned function will be called, therefore clear the timeout?